### DISCUSSION TITLE
Differences between PMV ASHRAE and EN

### DISCUSSION THREAD

**Discussion by Jacob-Knight** [CenterForTheBuiltEnvironment/comfort_tool#81] [General]
Thanks for the excellent CBE tool @FedericoTartarini @stefanoschiavon and team.  I have been looking at the difference between the ASHRAE PMV (ie with the calculated cooling effect) and the original EN version. As expected the ASHRAE version generally predicts more comfortable conditions at higher air speeds. 
![image](https://github.com/CenterForTheBuiltEnvironment/comfort_tool/assets/90765342/75a78591-2dc8-45a5-b44a-6329b7cbd0f8)
What is less intuitive is that it also predicts more comfortable conditions at higher Met values 
![image](https://github.com/CenterForTheBuiltEnvironment/comfort_tool/assets/90765342/12572c0d-e5b4-4e96-a23c-9a13cbdadbd2)

I think this is mostly due to the calculation assuming some self generated air movement at higher met values, which then in turn produces a significant cooling effect. I'm not totally convinced that represents reality, and was wondering if anyone was aware of research being done on this ? For example, someone exercising in a warm room at 25degC (To=25degC, rh = 50,  v = 0.15, clo=0.5 and met=3) gives a PMV (EN) of 1.7 (warm) but a PMV (ASHRAE) of 0.32 (neutral).  The reason the ASHRAE calc is so comfortable is that it has calculated a cooling effect of 8.5 degC which seems a bit excessive to me ? 

A possible factor in the cooling effect being so high is the function of the "CalculateCE" flag when calculating the SET. This is supposed to be "true" when doing the cooling effect calcs. However, if it is set to "false" when calculating the cooling effect, the cooling effect in the example above is significantly lower. Does anyone know the theoretical background to the "CalculateCE" flag - it seems to be saying that when calculating the cooling effect, you ignore the heat transfer coefficient due to convection at higher met values ("heatTransferConvMet" in the ASHRAE code), but I'm not sure why ?

A minor improvement to the CBE tool would be to print out the cooling effect temperature whenever the **relative** air speed is > 0.1m/s. At the moment it is only displayed when the (original) air speed is > 0.2m/s, so in the above example the tool is not telling you the value calculated.

**Comment by Jacob-Knight** [comment]
Just some more detail on what is happening with the above example To=25degC, rh = 50, v = 0.15, clo=0.5 and met=3
1) relative air speed calculated as 0.75m/s due to high met level
2) dynamic clothing calculated as 0.367 due to high met level
3) cooling effect calculation - the SET calculated with CalculateCE = true is 19.7degC (using the relative air speed and dynamic clothing). That then gives a very high cooling effect of 8.6degC
4) PMV calc done based on temp 25-8.6 = 16.4degC and air speed 0.1m/s
If the SET at stage 3 is calculated with CalculateCE = false, then the answer is 26.1degC, which would give a cooling effect of 0. 

An alternative question is whether it is correct to use the relative air speed and dynamic clo value when calculating the SET and cooling effect - ASHRAE D1 is not very clear on that. There is a rather cryptic note "For the use of SET in ASHRAE Standard 55, the function for self-generated air speed as a function of met rate has been removed" which I don't fully understand ? Perhaps the intention is that the acutal air speed (0.15 m/s in the example above) should be used in the SET and cooling effect calculations. In the above example, that would lead to: 
3) cooling effect calculation with CalculateCE = true is 27.2dec (using the actual air speed and dynamic clothing). That gives a cooling effect of 0.5 degC 
4) PMV calc ends up as 1.84 (warm) which is much closer to the PMV calculated with the EN method.
If that is how the calc is supposed to work, then there is a bug in the CBE tool and the place where it calls comf.cooling_effect needs to use the actual air speed not relative air speed
![image](https://github.com/CenterForTheBuiltEnvironment/comfort_tool/assets/90765342/122c5f34-92d2-41e3-ab8d-3b2e96488c64)

**Comment by FedericoTartarini** [comment]
Hi Jacob thank you for looking into this. I have tried to answer your comments below. I am also tagging @stefanoschiavon.

## Cooling effect with high met values
While in principle a cooling effect of 8C seems to be a bit high, I guess it also depends on the type of activity that the person is doing. I agree that the person would be feeling warm if performing an activity without moving from that spot (isometric exercise), on the other hand, I would not find it surprising if a person walking at to=25C, v=0.15m/s, rh=50%, met=3 (walking at 5 km/h), clo=0.5 would only have a PMV=0.32. This person would be moving into the air and this would cool the person down. However, since in buildings people would rarely walk around at this speed I agree that this result is a bit unexpected.

## CE flag
I would suggest you use [pythermalcomfort](https://pythermalcomfort.readthedocs.io/) if you are running multiple calculations. It is easier to use and the backend code is cleaner. The CBE thermal comfort tool and pythermalcomfort should output the same results. I prefer to run pythermalcomfort locally on my computer, however, you can also run it online in Google Colab. I am sharing a notebook that I created to calculate the cooling effect, you can access it [here](https://colab.research.google.com/drive/1nPWTJ8lyMNVxLD_kJ2rzVBxl1U3g3ULV?usp=sharing)

## Output the cooling effect
I agree, we should output the Cooling Effect value if v>0.1m/s. I will open an issue.

## Cooling effect calculation
The cooling effect adjustment to the PMV was introduced into an older version of the 55 Standard, but there is not a lot of documentation on how it should be implemented. I believe that the relative air speed should be used to calculate the SET. I consequently do not think that this is a bug in the tool, to the best of my knowledge this is how the PMV ASHRAE calculation should be implemented. But I may be wrong since the older version of the standard also did not provide very clear instructions on how it should be implemented. Ed Arens may know more about this, he is not on GitHub, so I am tagging @charliehuizenga who may know the answer.

Thank you again so much for looking into this. Hopefully, Stefano and Charlie will have more insights to share with you.

**Comment by Jacob-Knight** [comment]
Thanks @FedericoTartarini 
I agree it may not be a bug but it would be good if someone can comment on whether v or vr and clo or clo_d should be used for the cooling effect calcs. And if anyone knows the theory behind the "Calculate_CE" flag that would be helpful.
Using the above example again (met = 3)
SET (using v and clo, Calculate_CE = false) = 28.3 *this is the calc used for SET
SET (using v and clo, Calculate_CE = true) = 29.2
SET (using vr and clo_d, Calculate_CE = false) = 26.1
SET (using vr and clo_d, Calculate_CE = true) = 19.7  *this is the calc used as the basis of the cooling effect 
Trying to calculate the cooling effect using v and clo in this example produces a negative result (ie no cooling) which does suggest perhaps the current method is right, just an 8degC cooling effect does seem rather strange.

**Comment by FedericoTartarini** [comment]
The rationale for using vr and clo_d as inputs in the SET function is that the SET function is running within the PMV equation which uses vr and clo_d as inputs. 

Consequently, when the code runs in Python or in JavaScript, the SET function uses these adjusted values as inputs. It would not make much sense that a function that runs within the PMV code, uses inputs that are not passed to the parent function. 

I agree that a cooling of 8C seems to be very high. But to the best of my knowledge, this is how the PMV ASHRAE is designed to work. I would encourage you to use the original PMV instead if you are skeptical of the results. Stefano and Charlie may have more insights but they may be busy and they may not have time to respond.

**Reply by Jacob-Knight** [reply]
As I understand it the ASHRAE PMV calc on the CBE tool does the following (assuming air speed > 0.1):

- Calculates relative air speed vr 
- Calculates dynamic clo clo_d
- Calculates cooling effect using vr and clo_d  (this involves lots of SET calculations)
- Does the ISO PMV calc using temp = (Temp-cooling effect), air speed=0.1, and clo_d
- Does a SET calc for information using v and clo (this doesn't affect the PMV score)

I tend to agree that this is probably how it is meant to work, but it would be very good to have some official guidance from ASHRAE as to when clo/clo_d and v/vr should be used.

**Reply by FedericoTartarini** [reply]
You are correct. This is how it works. I agree that the ASHRAE 55 should provide more details on this.
